config/samples/ccruntime/base/ccruntime.yaml

```diff
@@ -64,6 +64,22 @@ spec:
             path: /etc/systemd/system/
             type: ""
           name: etc-systemd-system
+      environmentVariables:
+        # If set to true, this will install the CoCo fork of the containerd,
+        # the one allowing images to be pulled inside the guest and has patches
+        # for handling GPU / VFIO, on the node
+        # default: true
+        - name: "INSTALL_COCO_CONTAINERD"
+          value: "true"
+        # If set to true, this will install the v1.7.0 release of containerd on the node.
+        # default: false
+        - name: "INSTALL_OFFICIAL_CONTAINERD"
+          value: "false"
+        # If set to true, this will install the CoCo fork of the containerd,
+        # the one that has patches for handling GPU / VFIO, on the node
+        # default: false
+        - name: "INSTALL_VFIO_GPU_CONTAINERD"
```

Comment:
Do we really need three different options? Or just having INSTALL_COCO_CONTAINERD for custom containerd and INSTALL_OFFICIAL_CONTAINERD for officially released containerd version is good enough?

My apologies if this has been already discussed and I'm missing the full context.

Reply:
That's a very good point.

We have 3 different scenarios to cover, Pradipta.

1. We need to use the forked containerd, which allows us to pull the image on the guest.  This version is based on 1.6.x, and I really expect this to die as soon as possible, but that won't happen for v0.8.0 release as Enclave CC still depends on that.   Without much thinking, we also added to that containerd the VFIO / GPU related patches, as that was already a forked version of containerd.

2. We do not use the forked containerd, as we do rely on nydus-snapshotter to pull the image on the guest.
First of all, this brings in a requirement to use containerd v1.7.x, as that simplifies a lot the snapshotter setup, which can be done per runtime handler now, and that was not possible with 1.6.x.

Then within 2, we have the follow scenarios:
2.1. I just want to use it, I don't care about VFIO / GPUs:  Then the upstream containerd is what we want, and we can install it.  I hope we will never do that, to be honest, but we're giving the user the option to do so in order for them to easily test CoCo.

2.2. I do need to use VFIO / GPU stuff: Now we have to deal with non-upstreamed patches, and that's yet another containerd fork.  Could it be the same one as 1?  Maybe, but I'd prefer making it as clear as possible the differentiation of why we're using a forked version of containerd.  I'm afraid people will try to use VFIO / GPU and will end up just using the non-maintained / non-supported way of pulling image into the guest by mistake / lack of understanding on what has to be changed.

That's the reason we end up with those 3 different env vars. :-/

Reply:
Thanks @stevenhorsman @fidencio .. This helps ..:-)